custom_components/variable/sensor.py

```
@@ -458,38 +486,50 @@ async def async_increment_variable(self, **kwargs) -> None:
+                        self._attr_name,
+                        current_value,
                     )
-                    raise ValueError(f"Cannot convert current value to number: {current_value}")
+                    raise ValueError(
+                        f"Cannot convert current value to number: {current_value}"
+                    ) from None
+
+            if not isinstance(current_value, int | float):
+                raise TypeError(
+                    f"Cannot increment non-numeric value. Current value: {current_value}"
+                )
```

Ruff (0.16.1)

[warning] Abstract `raise` to an inner function

(TRY301)